Kernel/Arch/x86_64/PCI/Initializer.cpp

```diff
@@ -71,6 +72,11 @@ UNMAP_AFTER_INIT void initialize()
     PCIBusSysFSDirectory::initialize();
 
     MUST(PCI::enumerate([&](DeviceIdentifier const& device_identifier) {
+        // A simple sanity check to avoid getting a panic in get_interrupt_handler().
```

Comment:
The "sanity" check comment was for adding the `if` condition. But I can see it is a bit misleading. I will rephrase it.

> If we need to reserve the handler, can´t we do that in the PCI device initialization itself?

The previous commit message also has information regarding why we **don't** want to do it during device initialization. The main idea here is we need to mark an IRQ reserved as soon as possible because during the PCI device init, the driver may choose to use MSIx and we want to avoid sharing IRQs. Setting an IRQ as reserved during PCI device init is too late if we want to avoid sharing IRQ. Can I put this as a comment to make it clear?

> It also feels weird that we do this in the same loop where we output device information to the debug console

Should I create another loop to not tag along the debug print to make it more explicit?

Reply:
Gotcha. Looking at the change I might have overdone it a bit with the "it's weird to do it in the same loop" comment, but the code reads fine as it is now. Thanks!